File: app/src/main/java/tn/esprit/wayfinder/ui/components/TravelReelsFeed.kt
package tn.esprit.wayfinder.ui.components

import android.app.Application
import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Image
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavController
import coil.compose.AsyncImage
import androidx.compose.ui.res.painterResource
import tn.esprit.wayfinder.R
import tn.esprit.wayfinder.presentation.auth.ViewModelFactory
import tn.esprit.wayfinder.ui.theme.WayFinderSurface
import tn.esprit.wayfinder.utils.StringTranslator
import tn.esprit.wayfinder.viewmodels.ReelContentItem
import tn.esprit.wayfinder.viewmodels.ReelsViewModel
import tn.esprit.wayfinder.viewmodels.ReelsUiState

/**
 * Embedded preview section for reels feed in HomeScreen
 */
@Composable
fun TravelReelsFeed(
    navController: NavController,
    modifier: Modifier = Modifier
) {
    val context = LocalContext.current
    val reelsViewModel: ReelsViewModel = viewModel(
        factory = ViewModelFactory(context.applicationContext as Application)
    )
    val uiState by reelsViewModel.uiState.collectAsState()
    val colorScheme = MaterialTheme.colorScheme
    
    // Load reels on first composition
    LaunchedEffect(Unit) {
        reelsViewModel.loadReelsFeed(refresh = true)
    }
    
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 24.dp, vertical = 16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        // Section header
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = StringTranslator.translate(context, "Découvrez les voyages"),
                style = MaterialTheme.typography.titleLarge,
                fontWeight = FontWeight.Bold,
                color = colorScheme.onSurface
            )
            Text(
                text = StringTranslator.translate(context, "Voir tout"),
                style = MaterialTheme.typography.bodyMedium,
                color = Color(0xFF1976D2),
                modifier = Modifier.clickable {
                    navController.navigate("reels_viewer/0")
                }
            )
        }
        
        when (val state = uiState) {
            is ReelsUiState.Loading -> {
                // Loading skeleton
                LazyRow(
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                    contentPadding = PaddingValues(horizontal = 4.dp)
                ) {
                    items(3) {
                        ReelPreviewCardSkeleton()
                    }
                }
            }
            is ReelsUiState.Success -> {
                if (state.items.isEmpty()) {
                    // Empty state
                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        colors = CardDefaults.cardColors(
                            containerColor = WayFinderSurface
                        ),
                        shape = RoundedCornerShape(16.dp)
                    ) {
                        Box(
                            modifier = Modifier
                                .fillMaxWidth()
                                .height(200.dp)
                                .padding(24.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = StringTranslator.translate(context, "Aucun contenu disponible pour le moment"),
                                style = MaterialTheme.typography.bodyMedium,
                                color = colorScheme.onSurfaceVariant
                            )
                        }
                    }
                } else {
                    // Show preview cards (first 4 items)
                    val previewItems = state.items.take(4)
                    LazyRow(
                        horizontalArrangement = Arrangement.spacedBy(12.dp),
                        contentPadding = PaddingValues(horizontal = 4.dp)
                    ) {
                        items(previewItems.size) { index ->
                            val item = previewItems[index]
                            ReelPreviewCard(
                                item = item,
                                onClick = {
                                    navController.navigate("reels_viewer/$index")
                                }
                            )
                        }
                    }
                }
            }
            is ReelsUiState.Error -> {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    colors = CardDefaults.cardColors(
                        containerColor = colorScheme.errorContainer
                    ),
                    shape = RoundedCornerShape(16.dp)
                ) {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Text(
                            text = state.message,
                            style = MaterialTheme.typography.bodyMedium,
                            color = colorScheme.onErrorContainer
                        )
                        TextButton(onClick = { reelsViewModel.retry() }) {
                            Text(StringTranslator.translate(context, "Réessayer"))
                        }
                    }
                }
            }
            else -> {}
        }
    }
}

@Composable
fun ReelPreviewCard(
    item: ReelContentItem,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val colorScheme = MaterialTheme.colorScheme
    
    // Get thumbnail and content info
    val thumbnailUrl = when (item) {
        is ReelContentItem.PostItem -> item.post.imageUrl
        is ReelContentItem.JourneyItem -> {
            item.journey.videoUrl ?: item.journey.imageUrls.firstOrNull()
        }
    }
    
    val isVideo = when (item) {
        is ReelContentItem.PostItem -> false
        is ReelContentItem.JourneyItem -> item.journey.videoUrl != null
    }
    
    val creatorName = when (item) {
        is ReelContentItem.PostItem -> "${item.post.userId.firstName} ${item.post.userId.lastName}".trim()
        is ReelContentItem.JourneyItem -> item.journey.user?.let { 
            "${it.firstName ?: ""} ${it.lastName ?: ""}".trim().ifEmpty { it.username }
        } ?: "Traveler"
    }
    
    val creatorAvatar = when (item) {
        is ReelContentItem.PostItem -> item.post.userId.profileImageUrl
        is ReelContentItem.JourneyItem -> item.journey.user?.profileImageUrl
    }
    
    Card(
        modifier = modifier
            .width(160.dp)
            .height(240.dp)
            .clickable(onClick = onClick),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(
            containerColor = WayFinderSurface
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Box(modifier = Modifier.fillMaxSize()) {
            // Thumbnail/Image
            if (thumbnailUrl != null && thumbnailUrl.isNotBlank()) {
                val imageUrl = if (thumbnailUrl.startsWith("http")) {
                    thumbnailUrl
                } else {
                    "https://wayfinder-api-w92x.onrender.com$thumbnailUrl"
                }
                AsyncImage(
                    model = imageUrl,
                    contentDescription = null,
                    modifier = Modifier.fillMaxSize(),
                    contentScale = ContentScale.Crop,
                    placeholder = painterResource(id = R.drawable.europe),
                    error = painterResource(id = R.drawable.europe)
                )
            } else {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(
                            Brush.verticalGradient(
                                colors = listOf(
                                    Color(0xFF1976D2),
                                    Color(0xFF42A5F5)
                                )
                            )
                        )
                )
            }
            
            // Gradient overlay
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(
                        Brush.verticalGradient(
                            colors = listOf(
                                Color.Transparent,
                                Color.Black.copy(alpha = 0.6f)
                            ),
                            startY = 150f
                        )
                    )
            )
            
            // Video play icon
            if (isVideo) {
                Icon(
                    imageVector = Icons.Filled.PlayArrow,
                    contentDescription = "Play video",
                    tint = Color.White,
                    modifier = Modifier
                        .align(Alignment.Center)
                        .size(40.dp)
                        .background(Color.Black.copy(alpha = 0.5f), CircleShape)
                        .padding(8.dp)
                )
            }
            
            // Bottom content
            Column(
                modifier = Modifier
                    .align(Alignment.BottomStart)
                    .fillMaxWidth()
                    .padding(12.dp),
                verticalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                // Creator info
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    if (creatorAvatar != null && creatorAvatar.isNotBlank()) {
                        val avatarUrl = if (creatorAvatar.startsWith("http")) {
                            creatorAvatar
                        } else {
                            "https://wayfinder-api-w92x.onrender.com$creatorAvatar"
                        }
                        AsyncImage(
                            model = avatarUrl,
                            contentDescription = "Creator",
                            modifier = Modifier
                                .size(20.dp)
                                .clip(CircleShape),
                            contentScale = ContentScale.Crop,
                            placeholder = painterResource(id = R.drawable.europe),
                            error = painterResource(id = R.drawable.europe)
                        )
                    }
                    Text(
                        text = creatorName,
                        style = MaterialTheme.typography.bodySmall,
                        color = Color.White,
                        fontWeight = FontWeight.Medium,
                        maxLines = 1
                    )
                }
                
                // Engagement metrics
                Row(
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(4.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Filled.Favorite,
                            contentDescription = null,
                            tint = Color.White,
                            modifier = Modifier.size(12.dp)
                        )
                        Text(
                            text = formatCount(item.likesCount),
                            style = MaterialTheme.typography.bodySmall,
                            color = Color.White.copy(alpha = 0.9f),
                            fontSize = 11.sp
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun ReelPreviewCardSkeleton(
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier
            .width(160.dp)
            .height(240.dp),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant
        )
    ) {
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            CircularProgressIndicator(
                modifier = Modifier.size(32.dp),
                strokeWidth = 3.dp
            )
        }
    }
}

private fun formatCount(count: Int): String {
    return when {
        count >= 1_000_000 -> "${(count / 1_000_000f).let { if (it % 1 == 0f) it.toInt() else String.format("%.1f", it) }}M"
        count >= 1_000 -> "${(count / 1_000f).let { if (it % 1 == 0f) it.toInt() else String.format("%.1f", it) }}K"
        else -> count.toString()
    }
}

File: WayFinder/Views/Reels/TravelReelsFeed.swift
import SwiftUI

/// Embedded preview section for reels feed in HomeScreen
struct TravelReelsFeed: View {
    @StateObject private var viewModel = ReelsViewModel()
    @StateObject private var discussionViewModel = DiscussionViewModel()
    @Environment(\.colorScheme) private var colorScheme
    @Binding var navigateToReels: Bool
    @State private var showCreateReelSheet = false
    
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Section header
            HStack {
                Text("reels_discover_travels")
                    .font(.system(size: 22, weight: .bold))
                    .foregroundStyle(ThemeColors.primaryText(colorScheme))
                
                Spacer()
                
                Button(action: {
                    navigateToReels = true
                }) {
                    Text("reels_see_all")
                        .font(.body)
                        .foregroundStyle(Color(red: 0.098, green: 0.463, blue: 0.824))
                }
            }
            .padding(.horizontal, 24)
            
            switch viewModel.uiState {
            case .loading:
                // Loading skeleton
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 12) {
                        ForEach(0..<3, id: \.self) { _ in
                            ReelPreviewCardSkeleton()
                        }
                    }
                    .padding(.horizontal, 24)
                }
            case .success(let items, _):
                if items.isEmpty {
                    // Empty state
                    VStack(spacing: 16) {
                        Text("reels_no_content")
                            .font(.body)
                            .foregroundStyle(ThemeColors.secondaryText(colorScheme))
                            .multilineTextAlignment(.center)
                    }
                    .frame(maxWidth: .infinity)
                    .frame(height: 200)
                    .padding(.horizontal, 24)
                } else {
                    // Show preview cards (first 4 items) with Create Reel card as first item
                    let previewItems = Array(items.prefix(4))
                    ScrollView(.horizontal, showsIndicators: false) {
                        HStack(spacing: 12) {
                            // Create Reel Card (first item)
                            CreateReelCard(onTap: {
                                showCreateReelSheet = true
                            })
                            
                            // Other reel preview cards
                            ForEach(previewItems) { item in
                                ReelPreviewCard(
                                    item: item,
                                    onClick: {
                                        navigateToReels = true
                                    }
                                )
                            }
                        }
                        .padding(.horizontal, 24)
                    }
                }
            case .error(let message):
                VStack(spacing: 16) {
                    Text(message)
                        .font(.body)
                        .foregroundStyle(ThemeColors.secondaryText(colorScheme))
                        .multilineTextAlignment(.center)
                    
                    Button(action: {
                        viewModel.retry()
                    }) {
                        Text("reels_retry")
                            .font(.headline)
                            .foregroundColor(.white)
                            .padding(.horizontal, 24)
                            .padding(.vertical, 12)
                            .background(ThemeColors.accent())
                            .clipShape(Capsule())
                    }
                }
                .frame(maxWidth: .infinity)
                .frame(height: 200)
                .padding(.horizontal, 24)
            case .idle:
                EmptyView()
            }
            
            // Community Discussions Section (underneath reels)
            Spacer()
                .frame(height: 24)
            
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Text("home_community_discussions")
                        .font(.system(size: 22, weight: .bold))
                        .foregroundStyle(ThemeColors.primaryText(colorScheme))
                    
                    Spacer()
                    
                    Button(action: {
                        // Navigate to full discussions view
                    }) {
                        Text("reels_see_all")
                            .font(.body)
                            .foregroundStyle(Color(red: 0.098, green: 0.463, blue: 0.824))
                    }
                }
                .padding(.horizontal, 24)
                
                // Discussion posts preview
                DiscussionPostsPreview(viewModel: discussionViewModel)
                    .padding(.horizontal, 24)
            }
        }
        .task {
            await viewModel.loadReelsFeed(refresh: true)
            await discussionViewModel.loadPosts()
        }
        .sheet(isPresented: $showCreateReelSheet) {
            CreateReelSheet(
                onReelCreated: {
                    showCreateReelSheet = false
                    // Refresh reels feed
                    Task {
                        await viewModel.loadReelsFeed(refresh: true)
                    }
                },
                onDismiss: {
                    showCreateReelSheet = false
                }
            )
        }
    }
}

struct ReelPreviewCard: View {
    let item: ReelContentItem
    let onClick: () -> Void
    @Environment(\.colorScheme) private var colorScheme
    
    var body: some View {
        Button(action: onClick) {
            ZStack(alignment: .bottomLeading) {
                // Thumbnail/Image
                if let thumbnailUrl = item.thumbnailUrl, !thumbnailUrl.isEmpty {
                    AsyncImage(url: buildImageURL(from: thumbnailUrl)) { phase in
                        switch phase {
                        case .empty:
                            ProgressView()
                                .frame(width: 160, height: 240)
                        case .success(let image):
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fill)
                        case .failure:
                            gradientPlaceholder
                        @unknown default:
                            gradientPlaceholder
                        }
                    }
                    .frame(width: 160, height: 240)
                    .clipped()
                } else {
                    gradientPlaceholder
                        .frame(width: 160, height: 240)
                }
                
                // Gradient overlay
                LinearGradient(
                    colors: [
                        Color.clear,
                        Color.black.opacity(0.6)
                    ],
                    startPoint: .top,
                    endPoint: .bottom
                )
                .frame(width: 160, height: 240)
                
                // Video play icon
                if item.isVideo {
                    Image(systemName: "play.circle.fill")
                        .font(.system(size: 40))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity, maxHeight: .infinity)
                }
                
                // Bottom content
                VStack(alignment: .leading, spacing: 4) {
                    // Creator info
                    HStack(spacing: 6) {
                        if let avatarUrl = item.creatorAvatar, !avatarUrl.isEmpty, let url = buildImageURL(from: avatarUrl) {
                            AsyncImage(url: url) { phase in
                                switch phase {
                                case .empty:
                                    Circle()
                                        .fill(Color.white.opacity(0.3))
                                        .frame(width: 20, height: 20)
                                case .success(let image):
                                    image
                                        .resizable()
                                        .aspectRatio(contentMode: .fill)
                                case .failure:
                                    Circle()
                                        .fill(Color.white.opacity(0.3))
                                        .frame(width: 20, height: 20)
                                @unknown default:
                                    Circle()
                                        .fill(Color.white.opacity(0.3))
                                        .frame(width: 20, height: 20)
                                }
                            }
                            .frame(width: 20, height: 20)
                            .clipShape(Circle())
                        }
                        
                        Text(item.creatorName)
                            .font(.system(size: 12, weight: .medium))
                            .foregroundColor(.white)
                            .lineLimit(1)
                    }
                    
                    // Engagement metrics
                    HStack(spacing: 12) {
                        HStack(spacing: 4) {
                            Image(systemName: "heart.fill")
                                .font(.system(size: 12))
                                .foregroundColor(.white)
                            Text(formatCount(item.likesCount))
                                .font(.system(size: 11))
                                .foregroundColor(.white.opacity(0.9))
                        }
                        HStack(spacing: 4) {
                            Image(systemName: "eye.fill")
                                .font(.system(size: 12))
                                .foregroundColor(.white)
                            Text(formatCount(item.viewsCount))
                                .font(.system(size: 11))
                                .foregroundColor(.white.opacity(0.9))
                        }
                    }
                }
                .padding(12)
            }
        }
        .buttonStyle(.plain)
        .frame(width: 160, height: 240)
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .shadow(color: Color.black.opacity(colorScheme == .dark ? 0.2 : 0.05), radius: 4, x: 0, y: 2)
    }
    
    private var gradientPlaceholder: some View {
        LinearGradient(
            colors: [
                Color(red: 0.098, green: 0.463, blue: 0.824),
                Color(red: 0.259, green: 0.647, blue: 0.980)
            ],
            startPoint: .top,
            endPoint: .bottom
        )
    }
    
    private func buildImageURL(from urlString: String) -> URL? {
        if urlString.hasPrefix("http://") || urlString.hasPrefix("https://") {
            return URL(string: urlString)
        }
        let baseURL = "https://wayfinder-api-w92x.onrender.com"
        // Ensure urlString starts with / if it doesn't already
        let path = urlString.hasPrefix("/") ? urlString : "/\(urlString)"
        return URL(string: "\(baseURL)\(path)")
    }
    
    private func formatCount(_ count: Int) -> String {
        if count >= 1_000_000 {
            let value = Float(count) / 1_000_000.0
            return String(format: value.truncatingRemainder(dividingBy: 1) == 0 ? "%.0fM" : "%.1fM", value)
        } else if count >= 1_000 {
            let value = Float(count) / 1_000.0
            return String(format: value.truncatingRemainder(dividingBy: 1) == 0 ? "%.0fK" : "%.1fK", value)
        } else {
            return "\(count)"
        }
    }
}

struct ReelPreviewCardSkeleton: View {
    @Environment(\.colorScheme) private var colorScheme
    
    var body: some View {
        RoundedRectangle(cornerRadius: 16)
            .fill(ThemeColors.surface(colorScheme))
            .frame(width: 160, height: 240)
            .overlay {
                ProgressView()
            }
    }
}

struct DiscussionPostsPreview: View {
    @ObservedObject var viewModel: DiscussionViewModel
    @Environment(\.colorScheme) private var colorScheme
    
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            if viewModel.isLoading {
                ProgressView()
                    .frame(maxWidth: .infinity)
                    .padding()
            } else if let error = viewModel.errorMessage {
                Text(error)
                    .font(.caption)
                    .foregroundColor(.red)
                    .padding()
            } else if viewModel.posts.isEmpty {
                Text("No discussions yet")
                    .font(.caption)
                    .foregroundStyle(ThemeColors.secondaryText(colorScheme))
                    .frame(maxWidth: .infinity)
                    .padding()
            } else {
                ForEach(viewModel.posts.prefix(3)) { post in
                    CompactDiscussionPostCard(post: post)
                }
            }
        }
    }
}

struct CompactDiscussionPostCard: View {
    let post: DiscussionPost
    @Environment(\.colorScheme) private var colorScheme
    
    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 8) {
                if let avatarUrl = post.user.profileImageUrl, !avatarUrl.isEmpty {
                    AsyncImage(url: buildImageURL(from: avatarUrl)) { phase in
                        switch phase {
                        case .empty:
                            Circle()
                                .fill(Color.gray.opacity(0.3))
                                .frame(width: 32, height: 32)
                        case .success(let image):
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fill)
                        case .failure:
                            Circle()
                                .fill(Color.gray.opacity(0.3))
                                .frame(width: 32, height: 32)
                        @unknown default:
                            Circle()
                                .fill(Color.gray.opacity(0.3))
                                .frame(width: 32, height: 32)
                        }
                    }
                    .frame(width: 32, height: 32)
                    .clipShape(Circle())
                }
                
                VStack(alignment: .leading, spacing: 2) {
                    Text(post.user.username ?? "User")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundStyle(ThemeColors.primaryText(colorScheme))
                    
                    if let destination = post.destination {
                        Text(destination)
                            .font(.system(size: 12))
                            .foregroundStyle(ThemeColors.secondaryText(colorScheme))
                    }
                }
                
                Spacer()
            }
            
            Text(post.content)
                .font(.system(size: 14))
                .foregroundStyle(ThemeColors.primaryText(colorScheme))
                .lineLimit(2)
            
            HStack(spacing: 16) {
                HStack(spacing: 4) {
                    Image(systemName: "heart")
                        .font(.system(size: 12))
                    Text("\(post.likesCount)")
                        .font(.system(size: 12))
                }
                .foregroundStyle(ThemeColors.secondaryText(colorScheme))
                
                HStack(spacing: 4) {
                    Image(systemName: "bubble.right")
                        .font(.system(size: 12))
                    Text("\(post.commentsCount)")
                        .font(.system(size: 12))
                }
                .foregroundStyle(ThemeColors.secondaryText(colorScheme))
            }
        }
        .padding(12)
        .background(ThemeColors.surface(colorScheme))
        .clipShape(RoundedRectangle(cornerRadius: 12))
    }
    
    private func buildImageURL(from urlString: String) -> URL? {
        if urlString.hasPrefix("http://") || urlString.hasPrefix("https://") {
            return URL(string: urlString)
        }
        let baseURL = "https://wayfinder-api-w92x.onrender.com"
        let path = urlString.hasPrefix("/") ? urlString : "/\(urlString)"
        return URL(string: "\(baseURL)\(path)")
    }
}

